nwb_conversion_tools/utils/si013nwbephyswriter.py

```diff
@@ -0,0 +1,302 @@
+
+    @staticmethod
+    def supported_types():
+        assert HAVE_SI013
+        return (se.RecordingExtractor, se.SortingExtractor)
+
+    def get_num_segments(self):
+        return 1
+
+    def _get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled=True, segment_index=0):
+        return self.recording.get_traces(
+            channel_ids=channel_ids, start_frame=start_frame, end_frame=end_frame, return_scaled=return_scaled
+        ).T
+
+    def _get_dtype(self, return_scaled=True):
+        return self.recording.get_dtype(return_scaled)
+
+    def _get_channel_property_names(self):
+        property_names = set()
+        for chan_id in self._get_channel_ids():
+            for i in self.recording.get_channel_property_names(chan_id):
+                property_names.add(i)
+        return list(property_names)
+
+    def _fill_missing_property_values(self, ids, prop, get_prop_func):
+        self.dt_column_defaults = {list: [], str: "", Real: np.nan, np.ndarray: np.array([np.nan])}
+        # find the size of ndarray dtype:
+        for id in ids:
+            try:
+                id_data = get_prop_func(id, prop)
+                if isinstance(id_data, np.ndarray):
+                    self.dt_column_defaults.update({np.ndarray: np.nan * np.ones(shape=[1, id_data.shape[1:]])})
+                    break
+                else:
+                    break
+            except:
+                continue
+        # find the channel property dtype:
+        found_property_types = Real
+        for id in ids:
+            try:
+                id_data = get_prop_func(id, prop)
+                proptype = [proptype for proptype in self.dt_column_defaults if isinstance(id_data, proptype)]
+                if len(proptype) > 0:
+                    found_property_types = proptype[0]
+                    break
+                else:  # if property not found in the supported self.dt_column_defaults, then return None
+                    return
+            except:
+                continue
+        # build data array:
+        data = []
+        for id in ids:
+            try:
+                id_data = get_prop_func(id, prop)
+            except:
+                id_data = self.dt_column_defaults[found_property_types]
+            if found_property_types == Real:
+                data.append(np.float(id_data))
+            else:
+                data.append(id_data)
+        return np.array(data)
+
+    def _get_channel_property_values(self, prop):
+        if prop == "location":
+            return self.recording.get_channel_locations()
+        elif prop == "gain":
+            return self.recording.get_channel_gains()
+        elif prop == "offset":
+            return self.recording.get_channel_offsets()
+        elif prop == "group":
+            if any(["group_name" in self.recording._properties.get(id, "") for id in self.recording.get_channel_ids()]):
```

Comment:
Could we avoid accessing protected members of classes? `recording._properties` has a leading underscore, which indicates that it should not be accessed directly.

Reply:
thanks for catching that, yes we can directly use `recording.get_channel_property_names(id)` for this.